pkg/services/publicdashboards/validation/validator.go

```diff
@@ -0,0 +1,27 @@
+package validation
+
+import (
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/dashboards"
+	publicDashboardModels "github.com/grafana/grafana/pkg/services/publicdashboards/models"
+)
+
+func ValidateSavePublicDashboard(dto *publicDashboardModels.SavePublicDashboardConfigDTO, dashboard *models.Dashboard) error {
+	var err error
+
+	if len(dto.DashboardUid) == 0 {
+		return dashboards.ErrDashboardIdentifierNotSet
+	}
+
+	if hasTemplateVariables(dashboard) {
```

Comment:
I think we should retrieve the dashboard in this method. This allow us to also validate whether the dashboard exists or not before save

Reply:
Will make it so our new method to retrieve a dashboard returns an error if it cant be found